File: Assignment1/MST/src/RunExperiments.java
/*
CSE6140 HW1
This is an example of how your experiments should look like.
Feel free to use and modify the code below, or write your own experimental code, as long as it produces the desired output.
*/

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.PrintWriter;

public class RunExperiments{
	public static void main(String[] args){


		if (args.length < 3) {
			System.err.println("Unexpected number of command line arguments");
			System.exit(1);
		}

		String graph_file = args[0];
		String change_file = args[1];
		String output_file = args[2];

		PrintWriter output;
		output = new PrintWriter(output_file, "UTF-8");

		Graph G = parseEdges(graph_file);

		long startMST = System.nanoTime();
		int MSTweight = computeMST(G);
		long finishMST = System.nanoTime();

		//Subtract the start time from the finish time to get the actual algorithm running time
		double MSTtotal = (finishMST - startMST)/1000000;

		//Write to output file the initial MST weight and time
		output.println(Integer.toString(MSTweight) + " " + Double.toString(MSTtotal));

		//Iterate through changes file
		BufferedReader br = new BufferedReader(new FileReader(change_file));
		String line = br.readLine();
		String[] split = line.split(" ");
		int num_changes = Integer.parseInt(split[0]);
		int u, v, weight;

		while ((line = br.readLine()) != null) {
			split = line.split(" ");
			u = Integer.parseInt(split[0]);
			v = Integer.parseInt(split[1]);
			weight = Integer.parseInt(split[2]);

			//Run your recomputeMST function to recalculate the new weight of the MST given the addition of this new edge
			//Note: you are responsible for maintaining the MST in order to update the cost without recalculating the entire MST
			long start_newMST = System.nanoTime();
			int newMST_weight = recomputeMST(u,v,weight,G);
			long finish_newMST = System.nanoTime();

			double newMST_total = (finish_newMST - start_newMST)/1000000;

			//Write new MST weight and time to output file
			output.println(Integer.toString(newMST_weight) + " " + Double.toString(newMST_total));


		}

		output.close();
		br.close();




	}
}

File: Assignment1/MST/src/RunExperiments.cpp
/*
CSE6140 HW1
This is an example of how your experiments should look like.
Feel free to use and modify the code below, or write your own experimental code, as long as it produces the desired output.
*/
#include <iostream>
#include <vector>
#include <string>
#include <ctime>
#include <fstream>

using namespace std;

int main(int argc, char *argv[]) {

	/*
	1. inputs: graph file, change file, name of output file
	2. parseEdges to parse graph file
	3. calculate MST (returns integer, weight of MST); we print this integer to the output file
	4. loop through change file, call function pass in new edge and MST

	*/

	if (argc < 4) {
		cout << "Usage: " << argv[0] << " <graph_file> <change_file> <output_file>" << endl;
		return 1;
	}

	string graph_file = argv[1]; 
	string change_file = argv[2];
	string output_file = argv[3];

	ofstream output;
	output.open(output_file);

	//Write this function to parse edges from graph file to create your graph object
	Graph G = parseEdges(graph_file);

	//Run your MST function on graph G and collect as output the total weight of the MST
	clock_t startMST = clock();
	int MSTweight = computeMST(G);
	clock_t endMST = clock();

	//Subtract the start time from the finish time to get the actual algorithm running time
	clock_t totalTime = 1000 * (endMST - startMST) / (float) CLOCKS_PER_SEC;

	//Write initial MST weight and time to output file
	output << MSTweight << " " << totalTime << endl;

	//Iterate through changes file
	ifstream changes(change_file);
	
	int newMSTWeight = -1;

	if (changes.is_open()) {
		int numChanges;
		changes >> numChanges; //read number of changes

		int counter = 0;
		while (counter < numChanges) {
			int u, v, weight;
			changes >> u; //read u
			changes >> v; //read v
			changes >> weight; //read w(u,v)

			//Run your recomputeMST function to recalculate the new weight of the MST given the addition of this new edge
			//Note: you are responsible for maintaining the MST in order to update the cost without recalculating the entire MST
			clock_t startNewMST = clock();
			newMSTWeight = recomputeMST(u, v, weight, G);
			clock_t endNewMST = clock();

			clock_t totalNewMST = 1000 * (endNewMST - startNewMST) / (float) CLOCKS_PER_SEC;

			//Write new weight and time to output file
			output << newMSTWeight << " " << totalNewMST << endl;

			counter++;
		}
		changes.close();
	}

	output.close();

	return 0;

}